Gordon360/Controllers/LostAndFoundController.cs

```diff
@@ -0,0 +1,157 @@
+        [HttpGet]
+        [Route("missingitemsall")]
+        [StateYourBusiness(operation = Static.Names.Operation.READ_ALL, resource = Resource.LOST_AND_FOUND_MISSING_REPORT)]
+        public ActionResult<IEnumerable<MissingItemReportViewModel>> GetAllMissingItems()
+        {
+            IEnumerable<MissingItemReportViewModel> result = lostAndFoundService.GetMissingItemsAll();
+            if (result != null)
+            {
+                return Ok(result);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+
+        /// <summary>
+        /// Get the list of missing item reports for the currently authenticated user.
+        /// </summary>
+        /// <returns>ObjectResult - an http status code, with an array of MissingItem objects in the body </returns>
+        [HttpGet]
+        [Route("missingitems")]
+        public ActionResult<IEnumerable<MissingItemReportViewModel>> GetMissingItems()
+        {
+            var authenticatedUserUsername = AuthUtils.GetUsername(User);
+
+            IEnumerable<MissingItemReportViewModel> result = lostAndFoundService.GetMissingItems(authenticatedUserUsername);
+            if (result != null)
+            {
+                return Ok(result);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+        /// <summary>
+        /// Get a missing item report with given ID.
+        /// </summary>
+        /// <param name="id">The id of the report to get</param>
+        /// <returns>ObjectResult - an http status code, with a MissingItem object in the body </returns>
+        [HttpGet]
+        [Route("missingitemsbyid/{id}")]
+        public ActionResult<MissingItemReportViewModel> GetMissingItem(int id)
+        {
+            var authenticatedUserUsername = AuthUtils.GetUsername(User);
+
+            MissingItemReportViewModel? result = lostAndFoundService.GetMissingItem(id, authenticatedUserUsername);
+            if (result != null)
+            {
+                return Ok(result);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+        /// <summary>
+        /// Get all actions taken on a given missing item report.
+        /// </summary>
+        /// <param name="id">The id of the report to get</param>
+        /// <returns>ObjectResult - an http status code, with a list of Actions Taken objects </returns>
+        [HttpGet]
+        [Route(("missingitem/{id}/actionstakenall"))]
+        public ActionResult<IEnumerable<ActionsTakenViewModel>> GetActionsTaken(int id)
+        {
+            IEnumerable<ActionsTakenViewModel> result = lostAndFoundService.GetActionsTaken(id);
+            if (result != null)
+            {
+                return Ok(result);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
```

Comment:
Since there is no auth code here, it looks like anyone can get the actions taken for any missing item as long as they know (or guess) the missing item's Id. That seems like a significant privacy issue.

Reply:
Resolved in API #1111